feat: adds mTLS support in repo-server (#26715)
argoproj/argo-cd: docs/operator-manual/upgrading/3.4-3.5.md

For normal GitHub releases, **`sbom.tar.gz`** still includes **`bom-go-mod.spdx`** (Go dependencies, tag-value SPDX from `spdx-sbom-generator`) and **`bom-docker-image.spdx`** (the published release image, tag-value SPDX from `sigs.k8s.io/bom`). The UI dependency list is now **`bom-ui-pnpm.spdx.json`**: **SPDX 2.3 JSON** from **`pnpm sbom`**, replacing the old tag-value **`./ui`** output from `spdx-sbom-generator`.

If you consume this archive with tooling that only looked at **`*.spdx`** files, extend it to handle **`bom-ui-pnpm.spdx.json`** as well, or verify **`sbom.tar.gz`** using **`argocd-sbom.intoto.jsonl`** without depending on a fixed internal file list.

### mTLS support for repo-server

> Argo CD 3.5 introduced mutual TLS (mTLS) for the `argocd-repo-server`. As a reminder for operators upgrading to 3.5 and enabling mTLS (or maintaining an existing mTLS setup), note that all gRPC clients of the repo-server must be configured:

Reviewer: I think the above implies that upon upgrading to Argo CD 3.5, users have mandatory action items.
Is this clearer?
```suggestion
mTLS is opt-in and disabled by default — no action is required for operators who do not configure it. If you choose to enable mTLS by setting `--client-ca-path` on `argocd-repo-server`, all gRPC clients of the repo-server must also be configured with matching client certificates:
```

Author: I sack at documentation and I know it 😛

- `argocd-server` (API server)
- `argocd-application-controller`

> - `argocd-applicationset-controller`

Reviewer: ```suggestion
- `argocd-applicationset-controller`
- `argocd-notifications-controller`
```

Key points and flags:

- Repo server (server-side):
  - `--client-ca-path` enables mTLS by specifying the client CA bundle; cannot be combined with `--disable-tls`.
  - Server TLS cert/key are loaded from `/app/config/reposerver/tls/tls.crt` and `/app/config/reposerver/tls/tls.key` (overridable via `ARGOCD_APP_CONFIG_PATH`).

> - Clients (argocd-server, application-controller, applicationset-controller):

Reviewer: ```suggestion
- Clients (argocd-server, application-controller, applicationset-controller, notifications-controller):
```
